crud/crud.cpp

```diff
@@ -535,11 +535,16 @@ crud::handle_unsubscribe(const bzn::caller_id_t& /*caller_id*/, const database_m
 void
 crud::handle_create_db(const bzn::caller_id_t& caller_id, const database_msg& request, std::shared_ptr<bzn::session_base> session)
 {
-    bzn::storage_result result;
+    bzn::storage_result result{bzn::storage_result::ok};
 
     std::lock_guard<std::shared_mutex> lock(this->lock); // lock for write access
 
-    if (this->storage->has(PERMISSION_UUID, request.header().db_uuid()))
+    // if node is undefined here, there will be  segfault, when unit testing, make sure node is defined.
+    if (!this->node->bluzelle_public_key().empty() && (this->node->bluzelle_public_key() != caller_id))
+    {
+        result = bzn::storage_result::db_create_denied;
```

Comment:
Lets stick with access_denied.

Reply:
Done

Reply:
Removed the necessity for the node::owner_public_key method by adding an owner_public_key bzn::key_t member to crud - propogated this change to the rest of the code